index.rst

```diff
@@ -5,77 +5,331 @@
+We briefly review the existing functionality of the system and discuss the use case for the Catcher.
+The idea originated from the First-Look Analysis and Feedback Functionality (FAFF) working group, and a more in-depth analysis of the use cases can be found in their reports.
+
 
 Introduction
 ============
 
 One of the finding of the First-Look Analysis and Feedback Functionality (FAFF) working group :sitcomtn:`025` :cite:`SITCOMTN-025`, was the need for a service to ...
 
-   handle the low-level coordiation of identifying when a specific condition is met, then launching and monitoring an analysis process.
+   handle the low-level coordination of identifying when a specific condition is met, then launching and monitoring an analysis process.
 
-The original proposal from the FAFF working group was to develop a Commandable SAL Component (CSC), similar to the Watcher, that would provide be capable of monitoring data from the system and trigger certain actions (e.g. perform some computation and/or generate reports) based on some pre-defined conditions.
+The idea is basically to have a component capable of executing event-driven processing and report generation.
 
-Following additional discussions, we have decided to experiment with an alternative approach, replacing the CSC with an off-the-shelf solution provided by provided by Sasquatch (see :sqr:`068` :cite:`SQR-068`).
+The Rubin Observatory Control System currently contains a ``Watcher`` CSC that is capable of generating alerts from user defined rules (e.g. event-driven alerts).
+Moreover, the ``ScriptQueue`` and ``OCPS`` CSCs are both able to execute command-driven processes.
+While the ``ScriptQueue`` is tailored towards control system operations (e.g. moving the telescope, taking data, etc.), the ``OCPS`` is dedicated to executing DM-pipeline tasks processes.
+The missing piece of functionality is the capability of mixing a ``Watcher``-like behavior (event-processing) with ``ScriptQueue``/``OCPS``-like behavior.
+Furthermore, we are also interested in generating comprehensive reports from the results of these processes and alarms for particular conditions.
 
-As discussed in :sqr:`068` :cite:`SQR-068`, Sasquatch is based on InfluxDB OSS 2.x, a time series database. The new version of InfluxDB brings a number of important updates, amongst which two are relevant for stream processing; `Flux`_ and `InfluxDB task`_.
+The following are the key functionality we are looking for in the ``Catcher``.
 
-.. _Flux: https://docs.influxdata.com/flux/v0.x/
-.. _InfluxDB task: https://docs.influxdata.com/influxdb/v2.2/process-data/get-started/
+Commandable SAL Component
+-------------------------
 
+The ``Catcher`` needs to interact with the Rubin Observatory Control System, both by reading telemetry from the system and by commanding other CSCs (see :ref:`introduction-process-image-data`).
```

Comment:
Isn't the only CSC it commands the OCPS?
We might want to make it clear that the intention is not to have it automatically commanding CSCs that move hardware etc?

Reply:
Do we really want to be explicit about that though? If you think about it, the Catcher could be used in the future to provide some automation.

Reply:
meet in the middle and say:
... by commanding other CSCs, such as the OCPS.

I think it then says the original purpose but leaves the door open.